iot-hub/Samples/device/AndroidSample/things/build.gradle

```diff
@@ -0,0 +1,58 @@
+apply plugin: 'com.android.application'
+
+//***********************************************************************************************//
+def IOTHUB_DEVICE_CONNECTION_STRING_ENV_VAR_NAME = project.hasProperty('DeviceConnectionString') ? '"'+project.property('DeviceConnectionString')+'"' : '"Define Device Connection"'
+def STRING='String'
+//***********************************************************************************************//
+
+android {
+    compileSdkVersion 27
+
+
+    defaultConfig {
+        applicationId "android.iot.sdk.azure.microsoft.com.things"
+        minSdkVersion 24
+        targetSdkVersion 27
+        versionCode 1
+        versionName "1.0"
+
+        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
+
+        each {
+            //buildTypes.mBuildConfigFields 'DATATYPE','VARIABLE',|"GRADLE VARIABLE|"'
+            buildConfigField STRING, 'DeviceConnectionString', IOTHUB_DEVICE_CONNECTION_STRING_ENV_VAR_NAME
+        }
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+
+    packagingOptions {
+        exclude "META-INF/MSFTSIG.SF"
+        exclude "META-INF/MSFTSIG.RSA"
+        exclude 'META-INF/DEPENDENCIES'
+        exclude 'META-INF/NOTICE'
+        exclude 'META-INF/LICENSE'
+        exclude 'META-INF/LICENSE.txt'
+        exclude 'META-INF/NOTICE.txt'
+        exclude 'thirdpartynotice.txt'
+    }
+}
+
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation 'com.android.support.constraint:constraint-layout:1.1.3'
+    testImplementation 'junit:junit:4.12'
+    androidTestImplementation 'com.android.support.test:runner:1.0.2'
+    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
+    compileOnly 'com.google.android.things:androidthings:+'
+    implementation ('com.microsoft.azure.sdk.iot:iot-device-client:1.14.2'){
+        exclude module: 'azure-storage'
```

Comment:
Same comment here. Can you mark this as 1.14.2 and above ?

Reply:
We could do implementation ('com.microsoft.azure.sdk.iot:iot-device-client:1.14.+') or are you preferring implementation ('com.microsoft.azure.sdk.iot:iot-device-client:+') which basically means the latest SDK.  But the later is adding risk in breaking customer's app if we are introducing api changes...

Reply:
created a task to track on updating these for dynamic versioning.